File: csvToJson.py
import csv
import sys
import json

fieldnames=["visit_id","country","cluster_id","location.id","location.name","location.kind","location.coordinates[0]","location.coordinates[1]","quantifier.id","quantifier.initials"]

def convert(filename):
	csv_filename = filename[0]
	print "Opening CSV file: ",csv_filename 
	f=open(csv_filename, 'r')
	csv_reader = csv.DictReader(f,fieldnames)
	json_filename = csv_filename.split(".")[0]+".json"
	print "Saving JSON to file: ",json_filename
	jsonf = open(json_filename,'w')
	for r in csv_reader:
		data = json.dumps({"visit_id" : r['visit_id'], "country" : r['country'], "cluster_id" : r['cluster_id'], "location" : {"id": r['location.id'],
					"name" : r['location.name'],
     				"kind" : r['location.kind'],
     				"type" : "Point",
     				"coordinates" : [r['location.coordinates[0]'],r['location.coordinates[1]']]
					},
					"quantifier" : {"id": r['quantifier.id'],
					"initials" : r['quantifier.initials']}
    })
		jsonf.write(data) 
	f.close()
	jsonf.close()

if __name__=="__main__":
		convert(sys.argv[1:])

File: csv_to_json.rb
require 'csv'
require 'json'

file = File.open 'tist.csv', 'r'

outputs = []
10.times { outputs << [] }

CSV.parse(file) do |line|
  outputs.sample <<
    {
      visit_id: line[0],
      country: line[1],
      cluster_id: line[2],
      location: {
        id: line[3],
        name: line[4],
        kind: line[5],
        type: 'Point',
        coordinates: [line[6], line[7]]
      },
      quantifier: {
        id: line[8],
        initials: line[9]
      }
    }
end

outputs.each_with_index do |output, idx|
  File.write "tist/tist#{idx}.json", output.to_json
end
